File: types/thought.ts
export type EmotionType = 
  | 'anger'
  | 'anxiety'
  | 'sadness'
  | 'fear'
  | 'shame'
  | 'disappointment'
  | 'frustration'
  | 'guilt'
  | 'neutral'
  | 'joy'
  | 'gratitude'
  | 'calm'
  | 'hope';

export type EmotionCategory = 'negative' | 'positive' | 'neutral';

export interface Thought {
  id: string;
  content: string;
  emotion: EmotionType;
  category?: string;
  createdAt: number;
  reframe?: {
    content: string;
    createdAt: number;
    aiGenerated?: boolean;
  };
  tags?: string[];
  favorite?: boolean;
}

export interface MoodEntry {
  date: string; // YYYY-MM-DD format
  mood: number; // 1-5 scale
  emotions: EmotionType[];
  note?: string;
}

File: thoughtsreframer/thoughtsreframer/Models/Thought.swift
import Foundation

public struct Thought: Identifiable, Codable {
    public let id: UUID
    public var content: String
    public var reframedContent: String?
    public var emotion: String
    public var category: String
    public var moodRating: Int
    public let createdAt: Date
    public var updatedAt: Date
    public var isFavorite: Bool
    
    public init(content: String, emotion: String = "중립", category: String = "일반", moodRating: Int = 3) {
        self.id = UUID()
        self.content = content
        self.reframedContent = nil
        self.emotion = emotion
        self.category = category
        self.moodRating = moodRating
        self.createdAt = Date()
        self.updatedAt = Date()
        self.isFavorite = false
    }
}

public struct MoodEntry: Identifiable, Codable {
    public let id: UUID
    public let date: Date
    public let rating: Int
    public let emotions: [String]
    public let note: String?
    
    public init(rating: Int, emotions: [String], note: String? = nil) {
        self.id = UUID()
        self.date = Date()
        self.rating = rating
        self.emotions = emotions
        self.note = note
    }
} 